server/datastore/mysql/software.go

```diff
@@ -2882,301 +3113,601 @@ last_vpp_install AS (
+			} else {
+				softwareTitleStatment, softwareTitleArgs, err = sqlx.In(softwareTitleStatment, softwareTitleIds)
+			}
+			if err != nil {
+				return nil, nil, ctxerr.Wrap(ctx, err, "expand IN query for software titles")
+			}
+			softwareTitleStatment, softwareTitleArgsNamedArgs, err := sqlx.Named(softwareTitleStatment, namedArgs)
+			if err != nil {
+				return nil, nil, ctxerr.Wrap(ctx, err, "build named query for software titles")
+			}
+			softwareTitleStatment = strings.ReplaceAll(softwareTitleStatment, "AND true", matchClause)
+			args = append(args, softwareTitleArgsNamedArgs...)
+			args = append(args, softwareTitleArgs...)
+			if len(cveNamedArgs) > 0 {
+				args = append(args, cveNamedArgs...)
+			}
+			if len(cveMatchArgs) > 0 {
+				args = append(args, cveMatchArgs...)
+			}
+			if len(matchArgs) > 0 {
+				args = append(args, matchArgs...)
+			}
+			stmt += softwareTitleStatment
+		}
+
+		var vppVulnerableJoin string
+		if len(vppAdamIDs) > 0 {
+			if opts.VulnerableOnly || opts.ListOptions.MatchQuery != "" {
+				vppVulnerableJoin += " AND ( "
+				if !opts.VulnerableOnly && opts.ListOptions.MatchQuery != "" {
+					vppVulnerableJoin += `
+						NOT EXISTS (
+							SELECT 1
+							FROM
+								software
+							INNER JOIN 
+								software_cve ON software_cve.software_id = software.id
+							WHERE
+								software.title_id = software_titles.id
+						) OR
+					`
 				}
-				if hs.LastInstallInstalledAt != nil {
-					hs.SoftwarePackage.LastInstall.InstalledAt = *hs.LastInstallInstalledAt
+
+				vppVulnerableJoin += `
+				EXISTS (
+					SELECT 1
+					FROM
+						software
+					INNER JOIN 
+						software_cve ON software_cve.software_id = software.id
+				`
+				cveMetaJoin := "\nINNER JOIN cve_meta ON software_cve.cve = cve_meta.cve"
+
+				// Only join CVE table if there are filters
+				if hasCVEFilters {
+					vppVulnerableJoin += cveMetaJoin
 				}
+
+				vppVulnerableJoin += `
+					WHERE
+						software.title_id = software_titles.id
+				`
+				vppVulnerableJoin += cveMetaFilter
+				vppVulnerableJoin += "\n" + cveMatchClause
+				vppVulnerableJoin += "\n))"
 			}
 
-			// promote the last uninstall info to the proper destination fields
-			if hs.LastUninstallScriptExecutionID != nil && *hs.LastUninstallScriptExecutionID != "" {
-				hs.SoftwarePackage.LastUninstall = &fleet.HostSoftwareUninstall{
-					ExecutionID: *hs.LastUninstallScriptExecutionID,
-				}
-				if hs.LastUninstallUninstalledAt != nil {
-					hs.SoftwarePackage.LastUninstall.UninstalledAt = *hs.LastUninstallUninstalledAt
-				}
+			if len(softwareTitleIds) > 0 {
+				vppAdamStatment = ` UNION `
 			}
-		}
 
-		// promote the VPP app id and version to the proper destination fields
-		if hs.VPPAppAdamID != nil {
-			var version string
-			if hs.VPPAppVersion != nil {
-				version = *hs.VPPAppVersion
+			vppAdamStatment += `
+			-- SELECT for vpp apps
+			%s
+			FROM
+				software_titles
+			LEFT JOIN
```

Comment:
Why are we joining against installers here?

Reply:
🤦 good catch.